src/dataflow-types/src/client.rs

```diff
@@ -141,7 +140,13 @@ pub enum StorageCommand<T = mz_repr::Timestamp> {
     /// Create the enumerated sources, each associated with its identifier.
     ///
     /// For each identifier, there is a source description and a valid `since` frontier.
-    CreateSources(Vec<(GlobalId, (crate::types::sources::SourceDesc, Antichain<T>))>),
+    CreateSources(
+        Vec<(
+            GlobalId,
+            (crate::types::sources::SourceDesc, Antichain<T>),
+            Vec<(PartitionId, T, crate::sources::MzOffset)>,
```

Comment:
I think maybe this (the two highlighted lines; all other than `GlobalId`) should be one triple, as we are presenting "source key" and "stuff about the source". @benesch will certainly arrive to observe that the one triple it should be probably wants a name too, which I am happy to do after this lands if you would prefer not to.